src/Window.cpp

```diff
@@ -934,6 +996,18 @@ void Window::onPauseButtonPressed() {
     }
 }
 
+void Window::onRealButtonPressed() {
+    m_isReal = !m_isReal;
+    if (m_isReal) {
+        m_realButton->setText("Unrealistic");
+        real=true;
```

Comment:
No need to encode the state in a separate variable. You can just query the state of the button when you need it - pressed vs not.